src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,139 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private int maxSize = 10;
+    private int size;
+    private T[] array;
+    private T[] arrayGrow;
+
+    public ArrayList() {
+        array = (T[]) new Object[maxSize];
+    }
 
+    public void resize() {
+        if (size == maxSize) {
+            maxSize = (int) (maxSize * 1.5);
+            arrayGrow = (T[]) new Object[maxSize];
+            System.arraycopy(array, 0, arrayGrow, 0, size);
+            array = arrayGrow;
+        }
     }
 
     @Override
-    public void add(T value, int index) {
+    public void add(T value) {
+        resize();
+        for (int i = 0; i < size; i++) {
+            if ((array[i] != null && array[i] == value)
+                    || (array[i] != null && array[i].equals(value))) {
+                array[i] = value;
+                return;
+            }
+        }
+        array[size] = value;
+        size++;
```

Comment:
this is not java's ArrayList behavior. You need only the last two lines here, which can be combined into one BTW